drivers/soft_spi/soft_spi.c

```diff
@@ -0,0 +1,207 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_soft_spi
+ * @{
+ *
+ * @file
+ * @brief       Software SPI implementation
+ *
+ * @author      Markus Blechschmidt <Markus.Blechschmidt@haw-hamburg.de>
+ * @author      Peter Kietzmann     <peter.kietzmann@haw-hamburg.de>
+ */
+
+#include <stdio.h>
+#include <assert.h>
+
+#include "mutex.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#include "soft_spi.h"
+#include "soft_spi_params.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/**
+ * @brief   Allocate one lock per SPI device
+ */
+static mutex_t locks[1];
+
+static inline bool spi_bus_is_valid(spi_t bus)
+{
+    unsigned int spi_num = (unsigned int) bus;
+
+    if (sizeof(spi_config) / sizeof(spi_config[0]) < spi_num || spi_num < 0) {
+        return false;
+    }
+    return true;
+}
+
+void spi_init(spi_t bus)
+{
+    DEBUG("Soft SPI init\n");
+    /* This is left empty because the periph_init function will call this for
+       number of hardware spi ports */
+    assert(spi_bus_is_valid(bus));
+
+    /* initialize device lock */
+    mutex_init(&locks[bus]);
+    spi_init_pins(bus);
+}
+
+void spi_init_pins(spi_t bus)
+{
+    DEBUG("Soft SPI spi_init_pins\n");
+
+    assert(spi_bus_is_valid(bus));
+
+    /* check that miso is not mosi is not clk*/
+    assert(soft_spi_config[bus].mosi_pin != soft_spi_config[bus].miso_pin);
+    assert(soft_spi_config[bus].mosi_pin != soft_spi_config[bus].clk_pin);
+    assert(soft_spi_config[bus].miso_pin != soft_spi_config[bus].clk_pin);
+    /* mandatory pins */
+    assert((GPIO_UNDEF != soft_spi_config[bus].mosi_pin) || (GPIO_UNDEF != soft_spi_config[bus].miso_pin));
+    assert(GPIO_UNDEF != soft_spi_config[bus].clk_pin);
+
+    /* initialize clock pin */
+    gpio_init(soft_spi_config[bus].clk_pin, GPIO_OUT);
+    /* initialize optional pins */
+    if (GPIO_UNDEF != soft_spi_config[bus].mosi_pin) {
+        gpio_init(soft_spi_config[bus].mosi_pin, GPIO_OUT);
+        gpio_clear(soft_spi_config[bus].mosi_pin);
+    }
+}
+
+int spi_init_cs(spi_t bus, spi_cs_t cs)
+{
+    DEBUG("Soft SPI init CS\n");
+    if (!spi_bus_is_valid(bus)) {
+        DEBUG("Soft SPI bus not valid\n");
+        return SPI_NODEV;
+    }
+
+    if (cs == GPIO_UNDEF) {
+        if (soft_spi_config[bus].cs_pin == GPIO_UNDEF) {
+            DEBUG("Soft SPI no valid CS pin\n");
+            return SPI_NOCS;
+        }
+        DEBUG("Soft SPI set default CS line\n");
+        gpio_init(soft_spi_config[bus].cs_pin, GPIO_OUT);
+        gpio_set(soft_spi_config[bus].cs_pin);
+    }
+    else {
+        DEBUG("Soft SPI set user CS line\n");
+        gpio_init(cs, GPIO_OUT);
+        gpio_set(cs);
+    }
+
+    return SPI_OK;
+}
+
+int spi_acquire(spi_t bus, spi_cs_t cs, spi_mode_t mode, spi_clk_t clk)
+{
+    /* Right now we don't care about speed */
+    (void)clk;
```

Comment:
this should be also stated in the module description...

Reply:
Do you mean in the header file in 
```
/**
 * @defgroup    driver_soft_spi Soft SPI
 * @ingroup     drivers
 * @brief
```
?